File: VB/NavigationDemo.Wpf/ViewModels/NavigationPane/ContactsViewModel.vb
Imports DevExpress.Mvvm

Namespace NavigationDemo
    Public Class ContactsViewModel
        Implements ISupportParameter

        Public Overridable Property Parameter() As Object

        Public Overridable Property FirstName() As String
        Public Overridable Property LastName() As String
        Public Overridable Property Email() As String

        Public Overridable Sub AddContact()
            Dim viewModel = DirectCast(Parameter, ContactsNavigationViewModel)
            viewModel.Contacts.Add(New ContactItem() With {.FirstName = FirstName, .LastName = LastName, .Email = Email})
            FirstName = String.Empty
            LastName = String.Empty
            Email = String.Empty
        End Sub
        Public Overridable Function CanAddContact() As Boolean
            Return Not(String.IsNullOrEmpty(FirstName) OrElse String.IsNullOrEmpty(FirstName))
        End Function

        #Region "ISupportParameter"
        Private Property ISupportParameter_Parameter() As Object Implements ISupportParameter.Parameter
            Get
                Return Parameter
            End Get
            Set(ByVal value As Object)
                Parameter = value
            End Set
        End Property
        #End Region
    End Class
End Namespace

File: CS/NavigationDemo.Wpf/ViewModels/NavigationPane/ContactsViewModel.cs
using DevExpress.Mvvm;

namespace NavigationDemo {
    public class ContactsViewModel: ISupportParameter {
        public virtual object Parameter { get; set; }

        public virtual string FirstName { get; set; }
        public virtual string LastName { get; set; }
        public virtual string Email { get; set; }

        public virtual void AddContact() {
            var viewModel = (ContactsNavigationViewModel)Parameter;
            viewModel.Contacts.Add(new ContactItem() { FirstName = FirstName, LastName = LastName, Email = Email });
            FirstName = string.Empty;
            LastName = string.Empty;
            Email = string.Empty;
        }
        public virtual bool CanAddContact() {
            return !(string.IsNullOrEmpty(FirstName) || string.IsNullOrEmpty(FirstName));
        }

        #region ISupportParameter
        object ISupportParameter.Parameter {
            get { return Parameter; }
            set { Parameter = value; }
        }
        #endregion
    }
}
